### DISCUSSION TITLE
Ask for minutes - seconds for Expiration time in Lifecycle Configuration

### DISCUSSION THREAD

**Discussion by gmunumel** [minio/minio#15149] [Q&A] [answered]
Hello, 

We'd like to know or request for a feature so the Expiration time in Lifecycle Configuration could be set in minutes or seconds. 

In our case, we are creating a new version object per second. We want to delete / clean the old versions after 15 mins (for instance), so we could prevent minIO to increase so much.

We could perform a clean by ourselves but we don't know if could affect performance.  

![Screenshot 2022-06-22 110625](https://user-images.githubusercontent.com/276168/174990392-a7e651b6-f293-4d9c-95bb-b139438ae8f5.png)

**Comment by klauspost** [comment, marked as answer]
@gmunumel This is not possible. 

If you want transient objects and you are this tight on space, make your application clean up after itself - this will be instant and take just as many resources as ILM.

**Reply by gmunumel** [reply]
Hello @klauspost, thanks for your answer. 

I don't want to annoy, but why it isn't possible? It's because require a lot work, changes in architecture, etc?

**Reply by ravindk89** [reply]
It would break compatibility with S3 @gmunumel - you can review the [S3 PutBucketLifecycleConfiguration API docs](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketLifecycleConfiguration.html) as well, but the API specifically requires integers.

For us to support seconds would break compatibility with S3, which is something we try to avoid whenever possible. Beyond that, in our experience as Klaus noted, sub-day "expiration" is best handled on the application-side instead of depending on the MinIO Lifecycle Management scanner. The scanner is designed to be a system that does not threaten core performance/IOPS. Getting sub-day or down to second-level granularity isn't what the system is built for.

**Reply by harshavardhana** [reply]
> I don't want to annoy, but why it isn't possible? It's because require a lot work, changes in architecture, etc?

It is AWS S3 incompatible and it's our choice.

**Reply by klauspost** [reply]
@gmunumel First of all, this is limited by S3 API. Also, practically there is no way to implement minute-precision and make it scale in a reasonable fashion.

So you should not design your application to rely on such functionality.

**Reply by gmunumel** [reply]
Thanks guys for the detailed response. 

In that case I will implemented myself.
